### PR TITLE
NEXUS-23682: Implemented proxy endpoints in order to support new DB

### PR DESCRIPTION
**Pull request by igorudovika**
Implemented proxy endpoints in order to support new DB

### CODE REVIEW

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java [hunk 1]
```diff
@@ -65,6 +65,15 @@ public String getIcon() {
     return getValue(HelmProperties.ICON, String.class);
   }
 
+  public String getEgine() {
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java:68]: typo: `eNgine`

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java [hunk 2]
```diff
@@ -94,6 +103,18 @@ public void setAppVersion(final String appVersion) {
     attributesEnumMap.put(HelmProperties.APP_VERSION, appVersion);
   }
 
+  public void setEgine(final String engine) {
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java:106]: same typo

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+          // NOOP
+        }
+        else if (event instanceof CollectionEndEvent) {
+          rewrite = false;
+        }
+        emitter.emit(event);
+      }
+    }
+    catch (IOException ex) {
+      log.debug("Error rewriting urls in index.yaml", ex);
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java:75]: Isn't it better to use `log.error()` to log this exception?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        if (!fileName.isEmpty()) {
+          fileName = Paths.get(fileName).getFileName().toString();
+        }
+        scalarEvent = new ScalarEvent(scalarEvent.getAnchor(), scalarEvent.getTag(),
+            scalarEvent.getImplicit(), fileName, scalarEvent.getStartMark(),
+            scalarEvent.getEndMark(), scalarEvent.getStyle());
+      }
+    }
+    catch (URISyntaxException ex) {
+      log.debug("Invalid URI in index.yaml", ex);
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java:95]: Isn't it better to use `log.error()` to log this exception?

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.repository.helm.internal.content.recipe;
+
+import java.io.IOException;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java:18]: Unused import

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.sonatype.nexus.repository.cache.CacheInfo;
+import org.sonatype.nexus.repository.config.Configuration;
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java:23]: Unused import

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyRecipe.groovy [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+/**
+ * @since 1.0.10
+ */
+@Named(HelmProxyRecipe.NAME)
+@Singleton
+class HelmProxyRecipe
+    extends RecipeSupport
+{
+  public static final String NAME = 'helm-proxy'
+  public static final String PATH_NAME = 'path'
```
Comment (Darkvar) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyRecipe.groovy:68]: Not used variable

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java (hunk 1 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java]: Thanks, fixed 5aad649f943e777daa53bf0b4b845e75636af662

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java (hunk 2 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/HelmAttributes.java]: Thanks, fixed

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java (hunk 3 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java]: Thanks, fixed 5aad649f943e777daa53bf0b4b845e75636af662

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java (hunk 4 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/metadata/IndexYamlAbsoluteUrlRewriterSupport.java]: Thanks, fixed

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java (hunk 5 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java]: Thanks, fixed 5aad649f943e777daa53bf0b4b845e75636af662

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java (hunk 6 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java]: Thanks, fixed 5aad649f943e777daa53bf0b4b845e75636af662

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyRecipe.groovy (hunk 7 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyRecipe.groovy]: Thanks, fixed 5aad649f943e777daa53bf0b4b845e75636af662

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+  protected Content getCachedContent(final Context context)  throws IOException {
+    return content().getAsset(getUrl(context)).orElse(null);
+  }
+
+  @Override
+  protected Content store(final Context context, final Content content) throws IOException {
+    AssetKind assetKind = context.getAttributes().require(AssetKind.class);
+    switch (assetKind) {
+      case HELM_INDEX:
+        return new Content(content().putIndex(getUrl(context), content, assetKind));
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java:59]: You can just return the `Content` here

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+      case HELM_PACKAGE:
+        return new Content(content().putComponent(getUrl(context), content, assetKind));
```
Comment (chernikovmaksym) [inline nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java:61]: You can just return the `Content` here

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java (hunk 8 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java]: yep, fixed 79e201777af2036e80b3e0cc895f9d22c8238ed7

File: nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java (hunk 9 above)
Comment (igorudovika) [reply nexus-repository-helm/src/main/java/org/sonatype/repository/helm/internal/content/recipe/HelmProxyFacet.java]: yep, fixed 79e201777af2036e80b3e0cc895f9d22c8238ed7

**Pull request merged**